mypy/checkexpr.py

```diff
@@ -2041,6 +2047,41 @@ def check_union_call(self,
         return (make_simplified_union([res[0] for res in results]),
                 callee)
 
+    def analyze_getattr(self, callee: CallableType,
+                        args: List[Expression],
+                        arg_types: List[Type],
+                        arg_kinds: List[ArgKind],
+                        context: Context) -> Type:
+        if not (2 <= len(arg_types) <= 3):
+            return callee.ret_type  # Invalid usage, reported elsewhere
+
+        attr = try_getting_str_literals(args[1], arg_types[1])
+        if not attr:
+            return callee.ret_type
+
+        self_type = arg_types[0]
+        default = arg_types[2] if len(arg_types) == 3 else None
+        possible_types = []
+
+        for attr_literal in attr:
+            # When default is set, we don't need to report errors,
+            # even when attribute access is invalid.
+            msg = self.msg.clean_copy() if default is not None else self.msg
+            possible_types.append(analyze_member_access(
```

Comment:
Have you looked at the other `analyze_member_access` parameters, namely `in_literal_context`, `self_type`, `module_symbol_table`? Are the default values OK for them?

Reply:
Not closely, they might be off right now.